ILSANG/Sources/Views/Ranking/RankingViewModel.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  RankingViewModel.swift
+//  ILSANG
+//
+//  Created by Kim Andrew on 10/7/24.
+//
+
+import UIKit
+import SwiftUI
+import Alamofire
+
+class RankingViewModel: ObservableObject {
+    enum ViewStatus {
+        case error
+        case loading
+        case loaded
+    }
+    
+    @Published var viewStatus: ViewStatus = .loading
+    @Published var selectedXpStat: XpStat = .strength
+    @Published var userRank: [Rank] = []
+    @Published var mokData: [Rank] = [
+        Rank(xpType: "CHARM", xpPoint: 200, customerId: "1234-5678-91011", nickname: "TestUser1"),
+        Rank(xpType: "STRENGTH", xpPoint: 150, customerId: "2234-5678-91011", nickname: "TestUser2"),
+        Rank(xpType: "CHARM", xpPoint: 300, customerId: "3234-5678-91011", nickname: "TestUser3")
+    ]
+    
+    private let userNetwork: UserNetwork
+    
+    init(userNetwork: UserNetwork) {
+        self.userNetwork = userNetwork
+    }
+    
+    @MainActor
+    func changeViewStatus(_ viewStatus: ViewStatus) {
+        self.viewStatus = viewStatus
+    }
+    
+    @MainActor
+    func loadUserRank(xpStat: String) async {
+        let res = await userNetwork.getUserRank(xpstat: xpStat, page: 0)
```

Comment:
```suggestion
        let res = await userNetwork.getUserRank(xpstat: xpStat)
```

userNetwork.getUserRank() 리뷰남긴 부분 수정하고 여기 같이 수정하면 좋을 것 같습니다ㅏ!